src/language/dafnyInstallation.ts

```diff
@@ -231,24 +238,35 @@ export class DafnyInstaller {
         return false;
       }
     }
-    await this.execLog(`git clone --recurse-submodules ${LanguageServerConstants.DafnyGitUrl}`);
+
+    // Clone the right version
+    await this.execLog(`git clone -b v${configuredVersion} --depth 1 --recurse-submodules ${LanguageServerConstants.DafnyGitUrl}`);
     processChdir(Utils.joinPath(installationPath, 'dafny').fsPath);
-    await this.execLog('git fetch --all --tags');
-    await this.execLog(`git checkout v${configuredVersion}`);
-    await this.execLog('dotnet build Source/DafnyLanguageServer/DafnyLanguageServer.csproj');
+
+    const { path: dotnet } = await getDotnetExecutablePath();
+    // The DafnyCore.csproj has a few targets that call `dotnet` directly.
+    // If dotnet is configured in dafny.dotnetExecutablePath
+    // it MAY NOT be on the path.
+    // This will cause the build to fail.
+    // This works around this edge case.
+    const injectPath = `PATH=${path.dirname(dotnet)}:$PATH`;
```

Comment:
On Windows, the path separator is ";" not ":".... but since this code is not supposed to execute on Windows, I think it's ok.